### ISSUE TITLE
Titles and Subtiles in Waypointtutorials throw an error in Spigot 1.17

### ISSUE THREAD

**Issue by CloudeLecaw** [MartenM/ServerTutorialPlus#48]
May the title and subtitle management for waypointfindings when uses with STP need a little update.
When i start and normaly we set on our parts a title, a subtile and a firework..
none of them was displayed :§
```
[22:46:18] [Server thread/INFO]: Cloude_Lecaw issued server command: /angelteich
[22:46:18] [Server thread/INFO]: java.lang.ClassNotFoundException: net.minecraft.server.v1_17_R1.PacketPlayOutTitle
[22:46:18] [Server thread/WARN]: java.lang.NullPointerException: Cannot invoke "java.lang.Class.getConstructor(java.lang.Class[])" because the return value of "nl.martenm.servertutorialplus.helpers.NeedsReflection.getNMSClass(String)" is null
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.helpers.NeedsReflection.sendTitle(NeedsReflection.java:29)
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.ServerTutorialPoint.playDefault(ServerTutorialPoint.java:188)
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.ServerTutorialPoint$1.start(ServerTutorialPoint.java:81)
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.objects.TutorialController.start(TutorialController.java:78)
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.commands.ServerTutorialCommands.onCommand(ServerTutorialCommands.java:392)
[22:46:18] [Server thread/WARN]: 	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[22:46:18] [Server thread/WARN]: 	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149)
[22:46:18] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.dispatchCommand(CraftServer.java:764)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.network.PlayerConnection.handleCommand(PlayerConnection.java:1944)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1783)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1764)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:46)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:1)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:30)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.TickTask.run(SourceFile:18)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandler.executeTask(SourceFile:151)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandler.executeNext(SourceFile:125)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.bg(MinecraftServer.java:1125)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.executeNext(MinecraftServer.java:1118)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandler.awaitTasks(SourceFile:134)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.sleepForTick(MinecraftServer.java:1102)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1031)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:307)
[22:46:18] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:831)
[22:46:18] [Server thread/INFO]: java.lang.ClassNotFoundException: net.minecraft.server.v1_17_R1.PacketPlayOutTitle
[22:46:18] [Server thread/WARN]: java.lang.NullPointerException: Cannot invoke "java.lang.Class.getDeclaredClasses()" because the return value of "nl.martenm.servertutorialplus.helpers.NeedsReflection.getNMSClass(String)" is null
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.helpers.NeedsReflection.sendSubTitle(NeedsReflection.java:54)
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.ServerTutorialPoint.playDefault(ServerTutorialPoint.java:189)
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.ServerTutorialPoint$1.start(ServerTutorialPoint.java:81)
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.objects.TutorialController.start(TutorialController.java:78)
[22:46:18] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.commands.ServerTutorialCommands.onCommand(ServerTutorialCommands.java:392)
[22:46:18] [Server thread/WARN]: 	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[22:46:18] [Server thread/WARN]: 	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149)
[22:46:18] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.dispatchCommand(CraftServer.java:764)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.network.PlayerConnection.handleCommand(PlayerConnection.java:1944)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1783)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1764)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:46)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:1)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:30)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.TickTask.run(SourceFile:18)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandler.executeTask(SourceFile:151)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandler.executeNext(SourceFile:125)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.bg(MinecraftServer.java:1125)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.executeNext(MinecraftServer.java:1118)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.util.thread.IAsyncTaskHandler.awaitTasks(SourceFile:134)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.sleepForTick(MinecraftServer.java:1102)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1031)
[22:46:18] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:307)
[22:46:18] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:831)
[22:47:44] [Server thread/INFO]: java.lang.ClassNotFoundException: net.minecraft.server.v1_17_R1.PacketPlayOutTitle
[22:47:44] [Server thread/WARN]: java.lang.NullPointerException: Cannot invoke "java.lang.Class.getConstructor(java.lang.Class[])" because the return value of "nl.martenm.servertutorialplus.helpers.NeedsReflection.getNMSClass(String)" is null
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.helpers.NeedsReflection.sendTitle(NeedsReflection.java:29)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.ServerTutorialPoint.playDefault(ServerTutorialPoint.java:188)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.custom.CheckPoint.access$000(CheckPoint.java:32)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.custom.CheckPoint$1.start(CheckPoint.java:63)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.objects.TutorialController.finishPoint(TutorialController.java:139)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.custom.CheckPoint$1$1.run(CheckPoint.java:72)
[22:47:44] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftTask.run(CraftTask.java:81)
[22:47:44] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:400)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1252)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:436)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1200)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1027)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:307)
[22:47:44] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:831)
[22:47:44] [Server thread/INFO]: java.lang.ClassNotFoundException: net.minecraft.server.v1_17_R1.PacketPlayOutTitle
[22:47:44] [Server thread/WARN]: java.lang.NullPointerException: Cannot invoke "java.lang.Class.getDeclaredClasses()" because the return value of "nl.martenm.servertutorialplus.helpers.NeedsReflection.getNMSClass(String)" is null
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.helpers.NeedsReflection.sendSubTitle(NeedsReflection.java:54)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.ServerTutorialPoint.playDefault(ServerTutorialPoint.java:189)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.custom.CheckPoint.access$000(CheckPoint.java:32)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.custom.CheckPoint$1.start(CheckPoint.java:63)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.objects.TutorialController.finishPoint(TutorialController.java:139)
[22:47:44] [Server thread/WARN]: 	at nl.martenm.servertutorialplus.points.custom.CheckPoint$1$1.run(CheckPoint.java:72)
[22:47:44] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftTask.run(CraftTask.java:81)
[22:47:44] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_17_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:400)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1252)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:436)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1200)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1027)
[22:47:44] [Server thread/WARN]: 	at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:307)
[22:47:44] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:831)
```

**Comment by MartenM** [comment]
Looks like some things changed in the new update.
I will see if I can update this soon.

**Comment by CloudeLecaw** [comment]
I hope you might can find some time. Since we are allready on 1.17.1 :)

Well the tutorials work so ling.
the players just dont get the rockets and title infors that tells them to follow the particle circles :3

**Comment by MartenM** [comment]
Fixed by commit a6374c059632574c20a7a55afea12a60d304797d
Please let me know if there are any other issues.
